online/src/main/scala/ai/chronon/online/Fetcher.scala

```diff
@@ -190,23 +190,38 @@ class Fetcher(val kvStore: KVStore,
             val baseMap = internalMap ++ externalMap
             // used for derivation based on ts/ds
             val tsDsMap: Map[String, AnyRef] =
-              Map("ts" -> (requestTs).asInstanceOf[AnyRef], "ds" -> (requestDs).asInstanceOf[AnyRef])
-            val derivedMap: Map[String, AnyRef] = Try(
+              Map("ts" -> requestTs.asInstanceOf[AnyRef], "ds" -> requestDs.asInstanceOf[AnyRef])
+
+            val derivedMapTry: Try[Map[String, AnyRef]] = Try {
               joinCodec
                 .deriveFunc(internalResponse.request.keys, baseMap ++ tsDsMap)
                 .mapValues(_.asInstanceOf[AnyRef])
-                .toMap) match {
-              case Success(derivedMap) => derivedMap
-              case Failure(exception) => {
+            }
+
+            val (derivedMapCleaned, baseMapCleaned) = derivedMapTry match {
+              case Success(derivedMap) =>
+                val cleanedDerivedMap = derivedMap -- tsDsMap.keys
+                (cleanedDerivedMap, baseMap)
+              case Failure(exception) =>
                 ctx.incrementException(exception)
-                throw exception
-              }
+                val derivedExceptionMap =
+                  Map("derivation_fetch_exception" -> exception.traceString.asInstanceOf[AnyRef])
+                getJoinConf(joinName) match {
```

Comment:
Just curious, what is the purpose of calling getJoinConf here? In which case this function will return failure?

Reply:
we would have gotten the joinConf once already right? We should just use that. 

This could have updated meanwhile if the joinCodec has changed since the first fetch. Also that will make the code simpler to deal with.

Reply:
Got it. Because the fetch results from internal and external don't have the join config itself. 

but if we simply return baseMap anyways, we don't need to check if there is a star in derivations or not.

Reply:
Per the above comment, the join conf is not needed here. Getting the conf again can be deleted.